admin/code/LeftAndMain.php

```diff
@@ -180,6 +180,50 @@ class LeftAndMain extends Controller implements PermissionProvider {
 	protected $responseNegotiator;
 
 	/**
+	 * Gets the combined configuration of all LeafAndMain subclasses required by the client app.
+	 *
+	 * @return array
+	 *
+	 * WARNING: Experimental API
+	 */
+	public function getCombinedClientConfig() {
+		$combinedClientConfig = ['sections' => []];
+		$cmsClassNames = CMSMenu::get_cms_classes();
+
+		// Ordered by url_priority because the routing rules
+		// need to be registered in priority order on the client.
+		usort($cmsClassNames, function ($a, $b) {
+			$priorityA = Config::inst()->get($a, 'url_priority');
+			$priorityB = Config::inst()->get($b, 'url_priority');
+
+			if ($a == $b) {
```

Comment:
These are already sorted in AdminRootController::rules() method. I would just move the logic from there to just before the `return` in `CMSMenu::get_cms_classes`.